src/sfdxError.ts

```diff
@@ -231,6 +236,11 @@ export class SfdxError extends NamedError {
   // Additional data helpful for consumers of this error.  E.g., API call result
   public data: any; // tslint:disable-line:no-any
 
+  /**
+   * Some errors support `error.code` instead of `error.name`. This keeps backwards compatability.
+   */
+  public code: string;
```

Comment:
I guess the weird thing would be doing a set on name and code will still have the old value.

I wonder if making code a javascript style getter where it always returns the current value of name unless code is set itself.

Reply:
Good point. Ya, we could do that.

Reply:
The comment makes this sound like should be optional?  `public code?: string`